app/models/order.rb

```diff
@@ -21,8 +24,50 @@ def set_defaults
     self.state = "Created"
   end
 
+  def finished?
+    FINISHED_STATES.include?(state)
+  end
+
   def update_message(level, message)
     progress_messages << ProgressMessage.new(:level => level, :message => message, :tenant_id => tenant_id)
     touch
   end
+
+  def mark_approval_pending(msg = nil)
+    return if state == "Approval Pending"
+
+    mark_item(msg, :order_request_sent_at => Time.now.utc, :state => "Approval Pending")
```

Comment:
I think `order_request_sent_at` is meant for when we've actually sent the order through topology, so this might belong on line 45 instead of here.